Ryanyae project v2

ryanyae opened the pull request:
## Description

Started the making of ProjectV2.
- created new components and a wrapper for projects (still a nested router under the folder name ProjectV2 in the pages folder)
- Works similarly with the original project, but with less clutter of code a more readability.
- 
## Updates

[Please enter a brief summary of updates to this PR during code review - if any]

## Checklist

- [ ] Ran `npm run lint:fix` to format code
- [ ] Requested at least one reviewer
- [ ] Connected PR to Trello ticket (steps below)
- [ ] Addressed code review comments
- [ ] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

ryanyae commented on src/pages/ProjectV2/ProjectV2Wrapper.tsx:
```diff
@@ -0,0 +1,71 @@
+                    {
+                        // I wonder if its better to do ${currentProjectURL}/resources 
+                        // or /projectsV2/${match.params.project_id}/join as the redirecting URL
+                    }
```
Just want to highlight these comments

ryanyae replied:
starting to think that  /projectsV2/${match.params.project_id}/join maybe better just so that the code is more readable and the URL will be more intuitive.

ryanyae commented on src/pages/ProjectV2/ProjectV2Wrapper.tsx:
```diff
@@ -0,0 +1,71 @@
+                {
+                    //Browser router is better than Switch because it allows for nested routes
+                    // and allows for site navigation without refreshing the page
+                }
```
Just want to highlight this comment

ElisaHu1 commented on src/components/ProjectOverviewCard/ProjectOverviewCard.tsx:
```diff
@@ -4,7 +4,9 @@ import "./ProjectOverviewCard.css";
 const ProjectOverviewCard = ({project}: any) => {
 
     const navigateToProject = (projectName: String) => {
-        window.location.pathname = `${ROUTES.PROJECT.BASE}/${projectName}`
```
is this dead code?

ryanyae replied:
Do you mean if it will become dead in ProjectV2?

ElisaHu1 replied:
yeah i wonder why you comment it out in v1 version? I assume projecct overrview card is still been used in our current version right?

ryanyae replied:
I think I was trying to debug something and I forgot to revert this change

ElisaHu1 commented on src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:
```diff
@@ -0,0 +1,41 @@
+    return (
+        <div className="project-content-container" style={{ margin: '10rem' }}>
+            <div id="table-contents"
+            >
```
why the close params in a separate line?

ryanyae replied:
I'll fix this.

ElisaHu1 commented on src/pages/ProjectV2/Wrapper/ProjectV2Wrapper.tsx:
```diff
@@ -0,0 +1,41 @@
+const ProjectV2Wrapper: React.FC<ProjectProps> = ({ match }) => {
+
+    return (
+        <div className="project-content-container" style={{ margin: '10rem' }}>
```
the whole design of the wrapper is <table> sidebar and <project router> for content

ElisaHu1 replied:
nope, just to clarify, thanks!

ryanyae commented:
@ElisaHu1 I think I have addressed all the comments there, is there any more?

ElisaHu1 commented:
> @ElisaHu1 I think I have addressed all the comments there, is there any more?

what do you mean? I did not see you update this 
<img width="633" alt="image" src="https://github.com/UBC-VCL/VCL-content-platform/assets/104804980/2228f1b6-89d3-4a25-9a16-012282c42239">

ryanyae commented:
> > @ElisaHu1 I think I have addressed all the comments there, is there any more?
> 
> what do you mean? I did not see you update this <img alt="image" width="633" src="https://user-images.githubusercontent.com/104804980/282278913-2228f1b6-89d3-4a25-9a16-012282c42239.png">

Right, this is what I was forgetting.
